docs/improving-gci-experience.md

```diff
@@ -0,0 +1,377 @@
+# Introduction
+
+Whether you are a new or experienced GCI participant, a GCI mentor or a
+Zulip contributor who wants to know a bit more about our GCI experiences -
+welcome!
+
+This document is intended to summarize lessons learned over the first few
+weeks of Google Code-In at Zulip. It is also a shared reference of
+experiences and ideas on growing together for both students and mentors.
+As such it's open for further suggestions and improvements, to learn the
+most from diverse experiences.
+
+It also includes answers to many common questions and issues raised during
+code reviews and discussions on Zulip.
+[Social rules](#what-are-the-social-rules), [GCI tasks](#how-gci-tasks-work),
+and some ideas on [going beyond GCI](#how-do-i-go-beyond-gci-tasks)
+have been discussed here.
+
+# What are the social rules?
+
+As a Zulip contributor, you should adhere to our [community code of
+conduct](https://zulip.readthedocs.io/en/latest/code-of-conduct.html).
+There are also some GCI lessons learned:
+
+* Respect time and effort others are investing in their contributions and
+communication.
+
+* Be mindful of others' time - fellow participants are also waiting for their
+reviews in the task queue and mentors are managing multiple responsibilities
+while online.
+
+* Never assume malice - if you are waiting for review or an answer, it's
+usually because the other person is busy with their work, unavailable or
+waiting for you to answer their questions.
+
+* Be mindful of cultural differences - both participants and mentors come
+from different backgrounds (including countries, languages and cultures),
+please remember some nuances might be interpreted differently than you'd
+expect.
+
+* Please don't at-mention people - all of the mentors are reading through the
+feed as quickly as they can, while helping people.
```

Comment:
This should perhaps be softened a bit -- really we want "please don't @-mention people just to bug them to them to look at your task or help you with a problem they haven't already expressed interest in".  It's totally reasonable and normal to @-mention someone in a conversation.